internal/overlord/logstate/syslog_test.go

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) 2023 Canonical Ltd
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License version 3 as
+// published by the Free Software Foundation.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package logstate
+
+import (
+	"bufio"
+	"bytes"
+	"net"
+	"net/url"
+	"time"
+
+	"github.com/canonical/pebble/internal/servicelog"
+	. "gopkg.in/check.v1"
+)
+
+type syslogSuite struct{}
+
+var _ = Suite(&syslogSuite{})
+
+func (s *syslogSuite) TestEncodeEntry(c *C) {
+	client := &syslogClient{
+		version:        1,
+		pid:            "100146",
+		host:           "mycontainer",
+		msgid:          "-",
+		priority:       priorityVal(FacilityUserLevelMessage, SeverityInformational),
+		structuredData: syslogStructuredData("pebble", canonicalPrivEnterpriseNum, nil),
```

Comment:
Shall we add a test with a couple of structured data fields while we're here? (I realize we're not using them yet, but the code is there so we should probably have a test for it.)